homeassistant/config_entries.py

```diff
@@ -1004,6 +1015,27 @@ async def async_step_discovery(
         await self._async_handle_discovery_without_unique_id()
         return await self.async_step_user()
 
+    @callback
+    def async_abort(
+        self, *, reason: str, description_placeholders: Optional[Dict] = None
+    ) -> Dict[str, Any]:
+        """Abort the config flow."""
+        assert self.hass
+
+        # Remove reauth notification if no reauth flows are in progress
+        if not any(
```

Comment:
```suggestion
        if self.source == SOURCE_REAUTH and not any(
```

Reply:
tests are failing because of this addition. Do we fix all the tests, or do something else?

Reply:
@balloob thoughts? ^

Reply:
could we just do `if (self.context and self.source == SOURCE_REAUTH) and not any(...)`

Reply:
or maybe source can guard itself before using context

Reply:
`pylint` doesn't like that - I've set the default value of `context` to an empty dict which should get around the error. Happy to change this if needed

Reply:
I went with the guard approach - I misread this initially and didn't realize what you were suggesting, but this makes sense, thanks!